test/indexing.jl

```diff
@@ -27,8 +27,13 @@
         @test isa(ds.time, ReadOnlyArray)
         @test isa(ds.val1.time, ReadOnlyArray)
 
-        @test_throws ErrorException ds.time[3] = DateTime(2020, 1, 1)
-        @test_throws ErrorException ds.val1.time[3] = DateTime(2020, 1, 1)
+        if VERSION < v"1.8"
+            @test_throws ErrorException ds.time[3] = DateTime(2020, 1, 1)
+            @test_throws ErrorException ds.val1.time[3] = DateTime(2020, 1, 1)
+        else
+            @test_throws CanonicalIndexError ds.time[3] = DateTime(2020, 1, 1)
```

Comment:
I think this one is fine cause we're just testing whatever error is thrown by the `ReadOnlyArray`. It is unfortunate that we can just say throws an exception or maybe a parent type like `Exception`.